python/ray/util/client/worker.py

```diff
@@ -320,6 +334,27 @@ def _call_schedule_for_task(
                 logger.exception("Failed to deserialize {}".format(
                     ticket.error))
                 raise
+        self.num_tasks_scheduled += 1
+        self.total_outbound_message_size += task.ByteSize()
+        if not self.warning_raised and \
+                self.num_tasks_scheduled > TASK_WARNING_THRESHOLD:
+            logger.warning(
+                f"More than {TASK_WARNING_THRESHOLD} remote tasks have been "
+                "scheduled. This can be slow on Ray Client due to "
+                "communication overhead. If you're running many fine-grained "
+                "tasks consider batching them (details in the Ray Design "
+                "Pattern document).")
+            self.warning_raised = True
+        if not self.warning_raised and \
+                self.total_outbound_message_size > MESSAGE_SIZE_THRESHOLD:
+            logger.warning(
```

Comment:
Could we do:
```
if self.warning_raised:
   pass
elif ....
elif ....
```

Reply:
You should be able to use the `ray.util.debug.log_once` utility for this.

```
from ray.util.debug log_once

if log_once("client_communication_overhead"):
    logger.warning(...)
```

Reply:
Going with log_once since that's very nice and clean way to do it